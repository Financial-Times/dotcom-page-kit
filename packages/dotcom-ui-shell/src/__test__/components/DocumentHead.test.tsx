import React from 'react'
import renderer from 'react-test-renderer'
import Subject from '../../components/DocumentHead'

describe('dotcom-ui-shell/src/components/DocumentHead', () => {
  it('renders the document head', () => {
    const props = {
      description: 'Website description.',
      facebookPage: 'facebook-page-id',
      pageTitle: 'Page title',
      siteTitle: 'Website title',
      twitterSite: '@twitter_page',
      canonicalURL: 'https://my.site',
      metaTags: [{ rel: 'alternate', type: 'application/rss+xml', href: 'path/to/rss' }]
    }

    const tree = renderer.create(<Subject {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should not render smart banner meta tag when showSmartBanner false', () => {
    const props = {
      description: 'Website description.',
      facebookPage: 'facebook-page-id',
      pageTitle: 'Page title',
      siteTitle: 'Website title',
      twitterSite: '@twitter_page',
      canonicalURL: 'https://my.site',
      metaTags: [{ rel: 'alternate', type: 'application/rss+xml', href: 'path/to/rss' }],
      showSmartBanner: false
    }

    const tree = renderer.create(<Subject {...props} />).root
    expect(() => {
      tree.findByProps({name: 'apple-itunes-app'})
    }).toThrow()
  })
})
