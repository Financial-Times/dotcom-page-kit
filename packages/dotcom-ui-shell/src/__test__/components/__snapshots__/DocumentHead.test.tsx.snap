// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-shell/src/components/DocumentHead renders the document head 1`] = `
Array [
  <meta
    charSet="utf-8"
  />,
  <meta
    content="IE=edge"
    httpEquiv="X-UA-Compatible"
  />,
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />,
  <title>
    Page title | Website title
  </title>,
  <meta
    content="Website description."
    name="description"
  />,
  <link
    href="https://my.site"
    rel="canonical"
  />,
  <meta
    content="index,follow"
    name="robots"
  />,
  <meta
    content="4-t8sFaPvpO5FH_Gnw1dkM28CQepjzo8UjjAkdDflTw"
    name="google-site-verification"
  />,
  <meta
    href="path/to/rss"
    rel="alternate"
    type="application/rss+xml"
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"WebSite\\",\\"name\\":\\"Financial Times\\",\\"alternateName\\":\\"FT.com\\",\\"url\\":\\"http://www.ft.com\\"}",
      }
    }
    type="application/ld+json"
  />,
  <meta
    content="facebook-page-id"
    property="fb:pages"
  />,
  <meta
    content="@twitter_page"
    property="twitter:site"
  />,
  <link
    href="https://www.ft.com/__origami/service/image/v2/images/raw/ftlogo-v1%3Abrand-ft-logo-square-coloured?source=update-logos&width=32&height=32&format=png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="https://www.ft.com/__origami/service/image/v2/images/raw/ftlogo-v1%3Abrand-ft-logo-square-coloured?source=update-logos&width=194&height=194&format=png"
    rel="icon"
    sizes="194x194"
    type="image/png"
  />,
  <link
    href="https://www.ft.com/__origami/service/image/v2/images/raw/ftlogo-v1%3Abrand-ft-logo-square-coloured?source=update-logos&width=180&height=180&format=png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
]
`;
